### PR TITLE
[UPDATE] search serialization method for Hostel Notices

### PR DESCRIPTION
**Pull request by realsdx**
This PR changes the Serialization method for the search api, to support searching in two different models using same ViewSet.

Need to run `rebuild_index` command after merging this.

### CODE REVIEW

File: backend/search/serializers.py [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
-from academics.models import Notice
+from academics.models import Notice, HostelNotice
 from department.models import Faculty
 
 
-class NoticeSearchSerializer(serializers.ModelSerializer):
-    class Meta:
-        model = Notice
-        fields = ('title', 'notice_type', 'file', 'date')
+class GeneralNoticeSearchSerializer(serializers.Serializer):
```
Comment (Compro-Prasad) [inline backend/search/serializers.py]: This name would somehow also represent **General** notice type. So, the old name was good enough.

Comment (realsdx) [reply backend/search/serializers.py]: But, there are two models now for Notices, by convention previous one was suitable for Notice model

Comment (Compro-Prasad) [reply backend/search/serializers.py]: Keep it `NoticeSearchSerializer`. It looks more general than an explicit `GeneralNoticeSearchSerializer`.

Comment (realsdx) [reply backend/search/serializers.py]: Okay

**Comment by realsdx** [comment]
Merging now, 
Make sure to run `./manage.py rebuild_index` after updating

**Comment by Compro-Prasad** [comment]
@realsdx tag @hasibul236 for such updates.

**Pull request merged**